test/immutable-store-state.js

```diff
@@ -0,0 +1,40 @@
+import Alt from '../dist/alt-with-runtime'
+import Immutable from 'immutable'
+import { assert } from 'chai'
+import sinon from 'sinon'
+
+const alt = new Alt()
+
+const actions = alt.generateActions('fire')
+
+const assign = require('object-assign')
+
+const store = alt.createStore({
+  displayName: 'ImmutableStore',
+
+  state: Immutable.Map({}),
```

Comment:
@jdlehman Record has setter, but the main difference comparably with Map is that if you want to set value to key which wasn't defined in constructor, you'll get an exception.

Honestly, I don't think the it's good idea sticking to specific data structure.